src/packages.ts

```diff
@@ -154,22 +162,21 @@ function downloadFile(urlString: string, pkg: Package, logger: Logger, status: S
 
     return new Promise<void>((resolve, reject) => {
         if (!pkg.tmpFile || pkg.tmpFile.fd == 0) {
-            return reject(new Error("Package file unavailable"));
+            reject(new PackageError("Temporary package file unavailable", pkg));
+            return;
         }
 
         let request = https.request(options, response => {
             if (response.statusCode === 301 || response.statusCode === 302) {
                 // Redirect - download from new location
-                return downloadFile(response.headers.location, pkg, logger, status)
-                    .then(
-                        () => resolve(),
-                        err => reject(err));
+                resolve(downloadFile(response.headers.location, pkg, logger, status));
```

Comment:
Why this change? We're now calling resolve even if the `downloadFile` promise rejects.